Drivers/Inc/MotorController.h

```diff
@@ -0,0 +1,30 @@
+#ifndef __MOTORCONTROLLER_H
+#define __MOTORCONTROLLER_H
+
+#include <bsp.h>
+#include "CAN.h"
+
+/**
+ * @brief   Initializes the motor controller
+ * @param   None
+ * @return  None
+ */ 
+void MotorController_Init(void);
+
+/**
+ * @brief   Sends MOTOR DRIVE command on CAN2
+ * @param   newVelocity desired motor velocity setpoint in m/s
+ * @param   motorCurrent desired motor current setpoint as a percentage of max current setting
+ * @return  None
+ */ 
+void MotorController_Drive(uint32_t newVelocity, uint32_t motorCurrent);
```

Comment:
I think I've had this conversation with someone before, but I can't quite remember the answer from then. Which of these parameters takes priority? If we want some velocity _v_ and some current _i_, but in order to maintain _v_ we need to supply a current _i<sub>1</sub>_ > _i_, then what will the current be?

Reply:
There are 2 modes that the Tritium operates under. 
1. Torque control. During this mode, we always have to send a desired velocity that is unobtainable (some very large value). And the motor current is simply proportional to the percentage of the accelerator pressed. In that case, only the motor current is relevant.
2. Velocity control. During this mode, the tritium will use the maximum available current to try and achieve the desired velocity. The way the documentation reads, it seems like the current set point will take priority.